lib/sse.js

```diff
@@ -1,107 +1,175 @@
+var _ = require('lodash');
 var mongojs = require('mongojs');
 var inflect = require('i')();
-var _ = require('lodash');
+var tinySSE = require('tiny-sse');
 var Promise = require('bluebird');
-var sse = require('tiny-sse');
+var hl = require('highland');
+var JSONAPI_Error = require('./jsonapi-error');
 
-var SSE = function () {
+var SSE = function() {
 };
 
-SSE.prototype.init = function (router, adapter, options) {
-    this.router = router;
-    this.adapter = adapter;
-    this.options = options;
-    this.db = mongojs(options.oplogConnectionString);
-}
+SSE.prototype.init = function(config) {
+    this.config = config;
+    this.options = config.context.options;
+    this.harvesterApp = config.context;
+
+    //only listen to post events if the verb is not specified
+    this.verbs = config.verbs || ['post'];
 
+    //wraps it up in an array of single item, so that it fits the current logic without too many conditions
+    this.singleResourceName = config.singleResourceName && [config.singleResourceName];
+    this.db = mongojs(this.options.oplogConnectionString);
+    this.coll = this.db.collection('oplog.rs');
+    
+    var routePrefix = (config.singleResourceName) ? '/' + inflect.pluralize(config.singleResourceName) : '';
```

Comment:
Would be nice if you could use harvester's inflect option to decide if you need to inflect or not. Changes are coming that make this more stable.